calculator/src/main/java/edu/nextstep/camp/calculator/CalculatorViewModel.kt

```diff
@@ -23,6 +26,12 @@ class CalculatorViewModel(
     private val _errorEvent = MutableSharedFlow<Throwable>()
     val errorEvent = _errorEvent.asSharedFlow()
 
+    private val _expressionHistory = MutableStateFlow<List<ExpressionHistory>?>(null)
```

Comment:
```suggestion
    private val _expressionHistory = MutableStateFlow<List<ExpressionHistory>>(emptyList())
```
컬렉션은 null 대신 빈 컬렉션으로 처리하는 것을 권장해요.

Reply:
[ae5b264](https://github.com/next-step/android-calculator-mvvm/pull/113/commits/ae5b264cc6a09139886266e1f0450666204e29c7)
기존에는 null로 초기화를 해주는 경우가 많았는데 컬렉션은 빈 객체로 처리 감사합니다

한가지 궁금한게 있다면
StateFlow와 SharedFlow 각각 어떨때 사용하면 좋을까?에 대한 고민이 들곤 합니다.

Reply:
StateFlow와 SharedFlow의 차이점에 대해서는 특강으로 영직님이 잠깐 다뤄주신 것 같아요.
정리하자면, State는 상태를 다룰 경우에 사용하고 Shared는 이벤트를 다룰 경우에 사용합니다.
State는 Cold Flow이고 Shared는 Hot Flow입니다.
Cold와 Hot의 차이점은, Cold는 구독 시점에 데이터의 처음부터 끝까지를 collect하지만 Hot은 현재 시점의 데이터를 collect합니다. 예를 들면, Cold는 VOD, Hot은 실시간이라고 보시면 되겠습니다.